File: java/src/main/java/S005.java
import com.google.common.math.IntMath;

public class S005 {

    private static final int LIMIT = 20;

    public static long solve() {
        int ans = 1;
        for (int factor = 2; factor <= LIMIT; factor++) {
            int deficient = factor / IntMath.gcd(ans, factor);
            ans *= deficient;
        }
        return ans;
    }
}

File: cpp/S005.cpp
#include <iostream>
#include "maths.cpp"

using namespace std;

static const int LIMIT = 20;

int main() {
	int prod = 1;
	for (int i = 2; i <= LIMIT; i++) {
		int deficient = i / gcd(prod, i);
		prod *= deficient;
	}
	cout << prod << endl;
	return 0;
}
